### PR TITLE
new optional parameter for NexusPluginsVersion. fixes #5

### LINKED ISSUE #5: Create parameter for Nexus Repo version
Right now it's hard coded to an older version, we should probably create a parameter for this to make one less thing you need to change when it's done generating?

### PR DESCRIPTION
**Pull request by bhamail**
This PR adds an optional parameter that can specify which version of Nexus Repository Manager to build against.

While I was in there, I also added some default values for some other parameters.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,15 +36,19 @@ Generating a format plugin is as easy as running the following:
 
 Repeated for the "newline" challenged:
 
-    mvn archetype:generate -DarchetypeArtifactId=nexus-format-archetype -DarchetypeGroupId=org.sonatype.nexus.archetypes -DarchetypeVersion=1.0-SNAPSHOT -DgroupId=org.sonatype.nexus.repository -DartifactId=nexus-repository-foo -DpluginFormat=foo -DpluginClass=Foo -Dversion=0.0.1
+    mvn archetype:generate -DarchetypeArtifactId=nexus-format-archetype -DarchetypeGroupId=org.sonatype.nexus.archetypes -DarchetypeVersion=1.0-SNAPSHOT -DnexusPluginsVersion=3.18.0-01 -DgroupId=org.sonatype.nexus.repository -DartifactId=nexus-repository-foo -DpluginFormat=foo -DpluginClass=Foo -Dversion=0.0.1
```
Comment (doddi) [inline README.md]: It looks like `nexusPluginsVersion` is also optional, maybe add this to the optional parameters description below with default

File: src/main/resources/archetype-resources/pom.xml [hunk 2]
```diff
@@ -19,12 +19,12 @@
...
     <groupId>org.sonatype.nexus.plugins</groupId>
     <artifactId>nexus-plugins</artifactId>
-    <version>3.15.0-01</version>
+    <version>${nexusPluginsVersion}</version>
   </parent>
-  <groupId>${groupId}</groupId>
-  <artifactId>${artifactId}</artifactId>
-  <name>${project.groupId}:${project.artifactId}</name>
-  <version>${version}</version>
+  <groupId>\${groupId}</groupId>
```
Comment (doddi) [inline src/main/resources/archetype-resources/pom.xml]: Curious why the `\` are needed

Comment (bhamail) [reply src/main/resources/archetype-resources/pom.xml]: > Curious why the `\` are needed

Great question. It has to do with maven filtering voodoo: The \ character "escapes" the ${} that follows it. The result is the variable substitution string remains intact after maven resource filtering:

    <groupId>\${groupId}</groupId>

becomes:

    <groupId>${groupId}</groupId>

Which allows later variable substitution to to work when the Archetype is executed.
Prior to my changes, these resources where not being filtered by maven.

File: README.md (hunk 1 above)
Comment (bhamail) [reply README.md]: > It looks like `nexusPluginsVersion` is also optional, maybe add this to the optional parameters description below with default

I added more docs about the required and optional parameters. 
To minimize the number of places we reference hard coded versions I also doc'd a linked to the source that defines the `defaultNexusPluginsVersion`. Hopefully that helps. 
efd8234

**Pull request merged**